src/mapping/NearestNeighborGradientMapping.cpp

```diff
@@ -59,8 +59,8 @@ void NearestNeighborGradientMapping::mapConsistent(DataID inputDataID, DataID ou
   PRECICE_TRACE(inputDataID, outputDataID);
   precice::utils::Event e("map." + mappingNameShort + ".mapData.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
 
-  PRECICE_CHECK(input()->data(inputDataID)->hasGradient(), "Mesh \"{}\" does not contain gradient data. Using Nearest Neighbor Gradient requires gradient data for each vertices.",
-                "Check if hasGradient flag in the Data object was successfully initialized.",
+  PRECICE_CHECK(input()->data(inputDataID)->hasGradient(), "Mesh \"{}\" does not contain gradient data. Using Nearest Neighbor Gradient requires gradient data for all vertices. "
+                                                           "Check if hasGradient flag in the Data object was successfully initialized.",
```

Comment:
I don't understand the second sentence here:

> Check if hasGradient flag in the Data object was successfully initialized.

Is this sth the user can influence? Shouldn't this be an assertion instead?